packetbeat/protos/graphite/trans.go

```diff
@@ -0,0 +1,200 @@
+package graphite
+
+import (
+	"time"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/packetbeat/procs"
+	"github.com/elastic/beats/packetbeat/protos/applayer"
+)
+
+// Initialization constants
+type transactions struct {
+	config *transactionConfig
+
+	requests  messageList
+	responses messageList
+
+	onTransaction transactionHandler
+}
+
+type transactionConfig struct {
+	transactionTimeout time.Duration
+}
+
+type transactionHandler func(requ, resp *message) error
+
+// List of messages available for correlation
+type messageList struct {
+	head, tail *message
+}
+
+func (trans *transactions) init(c *transactionConfig, cb transactionHandler) {
+	trans.config = c
+	trans.onTransaction = cb
+}
+
+func (trans *transactions) onMessage(
+	tuple *common.IPPortTuple,
+	dir uint8,
+	msg *message,
+) error {
+	var err error
+
+	msg.Tuple = *tuple
+	msg.Transport = applayer.TransportTCP
+	msg.CmdlineTuple = procs.ProcWatcher.FindProcessesTuple(&msg.Tuple)
+
+	if msg.IsRequest {
+		if isDebug {
+			debugf("Received request with tuple: %s", tuple)
+		}
+		err = trans.onRequest(tuple, dir, msg)
+		err = trans.onResponse(tuple, dir, msg)
```

Comment:
the protocol clearly has no responses... no need to put the message into both lists and then have the correlation match a message with itself. I don't think the transaction type is required at all here. Drop the code doing the correlation and directly forward the message to `transPub.onMessage` (after renaming onTransaction to onMessage). The less code we have to maintain (even if its auto-generated), the better.